manta/src/main/java/ch/cyberduck/core/manta/MantaSession.java

```diff
@@ -0,0 +1,192 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import ch.cyberduck.core.AttributedList;
+import ch.cyberduck.core.Cache;
+import ch.cyberduck.core.Credentials;
+import ch.cyberduck.core.Host;
+import ch.cyberduck.core.HostKeyCallback;
+import ch.cyberduck.core.HostPasswordStore;
+import ch.cyberduck.core.ListProgressListener;
+import ch.cyberduck.core.LoginCallback;
+import ch.cyberduck.core.Path;
+import ch.cyberduck.core.UrlProvider;
+import ch.cyberduck.core.exception.AccessDeniedException;
+import ch.cyberduck.core.exception.BackgroundException;
+import ch.cyberduck.core.exception.LoginFailureException;
+import ch.cyberduck.core.features.AttributesFinder;
+import ch.cyberduck.core.features.Delete;
+import ch.cyberduck.core.features.Directory;
+import ch.cyberduck.core.features.Home;
+import ch.cyberduck.core.features.Move;
+import ch.cyberduck.core.features.MultipartWrite;
+import ch.cyberduck.core.features.Read;
+import ch.cyberduck.core.features.Touch;
+import ch.cyberduck.core.features.Write;
+import ch.cyberduck.core.ssl.DefaultX509KeyManager;
+import ch.cyberduck.core.ssl.DisabledX509TrustManager;
+import ch.cyberduck.core.ssl.SSLSession;
+import ch.cyberduck.core.ssl.ThreadLocalHostnameDelegatingTrustManager;
+import ch.cyberduck.core.ssl.X509KeyManager;
+import ch.cyberduck.core.ssl.X509TrustManager;
+import ch.cyberduck.core.threading.CancelCallback;
+
+import org.apache.commons.lang3.Validate;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+
+import com.joyent.manta.client.MantaClient;
+import com.joyent.manta.config.BaseChainedConfigContext;
+import com.joyent.manta.config.ChainedConfigContext;
+import com.joyent.manta.config.DefaultsConfigContext;
+import com.joyent.manta.config.SettableConfigContext;
+import com.joyent.manta.config.StandardConfigContext;
+
+public class MantaSession extends SSLSession<MantaClient> {
+
+    public static final Logger log = Logger.getLogger(MantaSession.class);
+
+    static final String HEADER_KEY_STORAGE_CLASS = "Durability-Level";
+
+    private final SettableConfigContext<BaseChainedConfigContext> config;
+
+    final MantaExceptionMappingService exceptionMapper;
+
+    private String keyFingerprint;
+
+    MantaPathMapper pathMapper;
+
+    public MantaSession(final Host h) {
+        this(h, new DisabledX509TrustManager(), new DefaultX509KeyManager());
+    }
+
+    public MantaSession(final Host h, final X509TrustManager trust, final X509KeyManager key) {
+        super(h, new ThreadLocalHostnameDelegatingTrustManager(trust, h.getHostname()), key);
+
+        pathMapper = new MantaPathMapper(this);
+        exceptionMapper = new MantaExceptionMappingService(this);
+        config = new ChainedConfigContext(
```

Comment:
Please extend from `HttpSession` to make `MantaClient` use the `HttpConnectionPoolBuilder` configured with custom trust and key managers.